modules/apps/data-engine/data-engine-rest-test/src/testIntegration/java/com/liferay/data/engine/rest/resource/v2_0/test/DataLayoutResourceTest.java

```diff
@@ -423,8 +472,52 @@ protected DataLayout testPutDataLayout_addDataLayout() throws Exception {
 			_dataDefinition.getId(), randomDataLayout());
 	}
 
+	private DataLayout _createRandomDataLayoutFieldsWithVisualProps() {
+		DataDefinitionField firstDataDefinitionField =
+			_dataDefinition.getDataDefinitionFields()[0];
+
+		String defaultLanguageId = _dataDefinition.getDefaultLanguageId();
+
+		Map<String, Object> dataLayoutFieldsWithVisualProps =
+			HashMapBuilder.<String, Object>put(
+				firstDataDefinitionField.getName(),
+				HashMapBuilder.<String, Object>put(
+					"predefinedValue",
+					HashMapBuilder.<String, Object>put(
+						defaultLanguageId, RandomTestUtil.randomString()
+					).build()
+				).put(
+					"tip",
+					HashMapBuilder.<String, Object>put(
+						defaultLanguageId, RandomTestUtil.randomString()
+					).build()
+				).put(
+					"label",
+					HashMapBuilder.<String, Object>put(
+						defaultLanguageId, RandomTestUtil.randomString()
+					).build()
+				).put(
+					"placeholder",
+					HashMapBuilder.<String, Object>put(
+						defaultLanguageId, RandomTestUtil.randomString()
+					).build()
+				).put(
+					"required", RandomTestUtil.randomBoolean()
+				).put(
+					"showLabel", RandomTestUtil.randomBoolean()
+				).build()
+			).build();
+
+		DataLayout randomDataLayoutWithVisualProps = randomDataLayout();
+
+		randomDataLayoutWithVisualProps.setDataLayoutFields(
+			dataLayoutFieldsWithVisualProps);
```

Comment:
Can you suppress dataLayoutFieldsWithVisualProps variable and put it inline? Thx

Reply:
Yup :+1: